### DISCUSSION TITLE
Use of option tests.<test>.steps.<step>.http.retries

### DISCUSSION THREAD

**Discussion by EstevoVazquez** [stepci/stepci#138] [Q&A] [answered]
I am trying to use the http.retries option in a test, but it doesn't seem to be working correctly. According to the documentation ( #https://docs.stepci.com/reference/workflow-syntax.html#tests-test-steps-step-http-retries), my understanding is that if the test fails (ckecks dont passed), it should retry the configured number of times. However, it seems that the test is only executed once. Here is an example of the test I am performing (which returns a 403 error until user permissions are setting via rabbit message in another micro service ). 

      - id: update
        name: Update resource
        http:
          retries: 2000000
          auth:
            $ref: "#/components/credentials/jwt"
          url: https://${{env.host}}/api/v1/resources/${{captures.creation_id}}
          method: PUT
          headers:
            Content-Type: application/json
            accept: application/json
          body: 
            file: ../request_bodies/resources/update.json
          check:
            status: 200

I think the parameter may not be working as expected, as the execution time with that retries value  is `Time:  1.663s, estimated 2s`

If someone could tell me how to use that parameter correctly, I would greatly appreciate it.

**Comment by mishushakov** [comment, marked as answer]
The retry option is only for retrying http requests that have received no response

**Comment by sebastianwis** [comment]
Hey Estevo @EstevoVazquez 
Thank you once again for your contribution here! 
It would be amazing to welcome you to our Discord if you haven't joined it yet: 
https://discord.com/invite/KqJJzJ3BTu
This is where we keep users up to date with updates, features and bug resolutions.
